yolov6/core/inferer.py

```diff
@@ -3,107 +3,186 @@
             img_paths = [source]  # files
         else:
-            raise Exception(f'Invalid path: {source}')
-        self.img_paths = [img_path for img_path in img_paths if img_path.split('.')[-1].lower() in IMG_FORMATS]
-
-    def infer(self, conf_thres, iou_thres, classes, agnostic_nms, max_det, save_dir, save_txt, save_img, hide_labels, hide_conf):
-        ''' Model Inference and results visualization '''
+            raise Exception(f"Invalid path: {source}")
+        self.img_paths = [
+            img_path
+            for img_path in img_paths
+            if img_path.split(".")[-1].lower() in IMG_FORMATS
+        ]
+
+    def infer(
+        self,
+        conf_thres,
+        iou_thres,
+        classes,
+        agnostic_nms,
+        max_det,
+        save_dir,
+        save_txt,
+        save_img,
+        hide_labels,
+        hide_conf,
+    ):
+        """Model Inference and results visualization"""
 
         for img_path in tqdm(self.img_paths):
-            img, img_src = self.precess_image(img_path, self.img_size, self.stride, self.half)
+            img, img_src = self.precess_image(
+                img_path, self.img_size, self.stride, self.half
+            )
             img = img.to(self.device)
             if len(img.shape) == 3:
                 img = img[None]
                 # expand for batch dim
             pred_results = self.model(img)
-            det = non_max_suppression(pred_results, conf_thres, iou_thres, classes, agnostic_nms, max_det=max_det)[0]
+            det = non_max_suppression(
+                pred_results,
+                conf_thres,
+                iou_thres,
+                classes,
+                agnostic_nms,
+                max_det=max_det,
+            )[0]
 
             save_path = osp.join(save_dir, osp.basename(img_path))  # im.jpg
-            txt_path = osp.join(save_dir, 'labels', osp.basename(img_path).split('.')[0])
+            txt_path = osp.join(
+                save_dir, "labels", osp.basename(img_path).split(".")[0]
+            )
 
             gn = torch.tensor(img_src.shape)[[1, 0, 1, 0]]  # normalization gain whwh
             img_ori = img_src
 
             # check image and font
-            assert img_ori.data.contiguous, 'Image needs to be contiguous. Please apply to input images with np.ascontiguousarray(im).'
+            assert (
+                img_ori.data.contiguous
+            ), "Image needs to be contiguous. Please apply to input images with np.ascontiguousarray(im)."
             self.font_check()
 
             if len(det):
-                det[:, :4] = self.rescale(img.shape[2:], det[:, :4], img_src.shape).round()
+                det[:, :4] = self.rescale(
+                    img.shape[2:], det[:, :4], img_src.shape
+                ).round()
+
+                if isinstance(self.inference_logger, WandbInferenceLogger):
+                    self.inference_logger.in_infer(np.array(img_ori), save_path, reversed(det))
```

Comment:
What's the right thing to do here ? 
1. Should we pass the img_path and not the save_path ?
2. Why pass both the image and the image_path ? Wandb.Image can read a path.
3.  Passing only the img_path should also avoid the cv2 import in the wandb_inference_logger.py that is explicitly used only to convert BRG to RGB

Reply:
@parambharat 

> Should we pass the img_path and not the save_path

Done.

> Why pass both the image and the image_path ? Wandb.Image can read a path

We still need the dimensions of the image in order to normalize the bounding box coordinates.

>Passing only the img_path should also avoid the cv2 import in the wandb_inference_logger.py that is explicitly used only to convert BRG to RGB

I'm passing the imagr because we need the height and width of the image. However, passing `img_path` instead of `save_path` eliminates the need for `import cv2`.